Data/ByteString/Streaming/Char8.hs

```diff
@@ -710,20 +711,215 @@ putStrLn :: MonadIO m => ByteString m r -> m r
 putStrLn bs = hPut IO.stdout (snoc bs '\n')
 {-# INLINE putStrLn #-}
 
--- | This will read positive or negative Ints that require 18 or fewer characters.
-readInt :: Monad m => ByteString m r -> m (Compose (Of (Maybe Int)) (ByteString m) r)
-readInt = go . toStrict . splitAt 18 where
-  go m = do
-    (bs :> rest) <- m
-    case Char8.readInt bs of
-      Nothing -> return (Compose (Nothing :> (chunk bs >> rest)))
-      Just (n,more) -> if B.null more
-        then do
-          e <- uncons rest
-          return $ case e of
-            Left r -> Compose (Just n :> return r)
-            Right (c,rest') -> if isDigit c
-               then Compose (Nothing :> (chunk bs >> cons' c rest'))
-               else Compose (Just n :> (chunk more >> cons' c rest'))
-        else return (Compose (Just n :> (chunk more >> rest)))
+-- | Bounds for Word# multiplication by 10 without overflow, and
+-- absolute values of Int bounds.
+intmaxWord, intminWord, intmaxQuot10, intmaxRem10, intminQuot10, intminRem10 :: Word
+intmaxWord = fromIntegral (maxBound :: Int)
+intminWord = fromIntegral (negate (minBound :: Int))
+(intmaxQuot10, intmaxRem10) = intmaxWord `quotRem` 10
+(intminQuot10, intminRem10) = intminWord `quotRem` 10
+
+-- Predicate to test whether a 'Word8' value is either ASCII whitespace,
+-- or a unicode NBSP (U+00A0).  Optimised for ASCII text, with spaces
+-- as the most frequent whitespace characters.
+w8IsSpace :: Word8 -> Bool
+w8IsSpace = \ !w8 ->
+    -- Avoid the cost of narrowing arithmetic results to Word8,
+    -- the conversion from Word8 to Word is free.
+    let w :: Word
+        !w = fromIntegral w8
+     in w - 0x21 > 0x7e   -- not [x21..0x9f]
+        && ( w == 0x20    -- SP
+          || w - 0x09 < 5 -- HT, NL, VT, FF, CR
+          || w == 0xa0 )  -- NBSP
+{-# INLINE w8IsSpace #-}
+
+-- Try to position the stream at the next non-whitespace input, by
```

Comment:
```suggestion
-- | Try to position the stream at the next non-whitespace input, by
```

Reply:
Thanks, pushed a fixup.  Once all the reviews are done, I'd like to squash all the commits before this is merged.